### PR TITLE
fea(): fixed pytest errors for gptneox

### PR DESCRIPTION
**Pull request by imangohari1**
# What does this PR do?
fixes the generic error
```
FAILED tests/transformers/tests/models/gpt_neox/test_modeling_gpt_neox.py::GPTNeoXModelTest::test_retain_grad_hidden_states_attentions - TypeError: GPTNeoXSdpaAttention.forward() got an unexpected keyword argument 'token_idx'
```
tested locally and it works as expcted. 

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by ssarkar2** [comment]
sample: `RUN_SLOW=true  GAUDI2_CI=1 python -m pytest tests/transformers/tests/models/gpt_neox/test_modeling_gpt_neox.py  -v -s -k test_model_for_question_answering`

**Comment by imangohari1** [comment]
```
RUN_SLOW=true  GAUDI2_CI=1 python -m pytest tests/transformers/tests/models/gpt_neox/test_modeling_gpt_neox.py  -v -s 
```
```
=========================================== short test summary info =======================================================================================================
FAILED tests/transformers/tests/models/gpt_neox/test_modeling_gpt_neox.py::GPTNeoXModelTest::test_generate_from_inputs_embeds_decoder_only - RuntimeError: The input size 2, plus negative padding 0 and -44 resulted in a invalid output size, Check dimension 0 of your input.
================================================================================== 1 failed, 64 passed, 21 skipped, 6 warnings in 155.59s (0:02:35) =========
```


> sample: `RUN_SLOW=true GAUDI2_CI=1 python -m pytest tests/transformers/tests/models/gpt_neox/test_modeling_gpt_neox.py -v -s -k test_model_for_question_answering`

File: optimum/habana/transformers/modeling_utils.py [hunk 1]
```diff
@@ -356,6 +356,7 @@ def adapt_transformers_to_gaudi():
     transformers.models.gpt_neox.modeling_gpt_neox.GPTNeoXModel.forward = gaudi_gpt_neox_model_forward
     transformers.models.gpt_neox.modeling_gpt_neox.GPTNeoXLayer.forward = gaudi_gpt_neox_layer_forward
     transformers.models.gpt_neox.modeling_gpt_neox.GPTNeoXAttention.forward = gaudi_gpt_neox_attention_forward
+    transformers.models.gpt_neox.modeling_gpt_neox.GPTNeoXSdpaAttention.forward = gaudi_gpt_neox_attention_forward
```
Comment (regisss) [inline optimum/habana/transformers/modeling_utils.py:359]: Not a big fan of doing this. This change should be enough: https://github.com/huggingface/optimum-habana/pull/1163/files#diff-bae02284f455b93397dbeafee178bd779671429602246f3ba60ea833a538eb68R297
Unless the test doesn't use `GPTNeoXForCausalLM`? Which test is it?

Comment (regisss) [reply optimum/habana/transformers/modeling_utils.py:359]: Just saw the test above, let me check

Comment (regisss) [reply optimum/habana/transformers/modeling_utils.py:359]: I pushed this commit to do it differently, the test above still passes

**Pull request merged**